### ISSUE TITLE
Carbs -> Bolus screen. back out

### ISSUE THREAD

**Issue by CSlycord1** [Artificial-Pancreas/iAPS#32]
Not an issue per se but more of a requested function.

When adding carbs and moving to the bolus screen.  If for some reason you want to back out of the bolus screen, it would be nice if iAPS would confirm if you still want to log the carbs.  Current state is that when you back out (close) the bolus screen carbs are still logged.  Deleting the carb entry is easy enough if one knows of this "bug" , but a more frustrating thing is if you have plugged in protein and fat.  The equivalent carb entries are broken out into singular carb entries over many hours.

I can break this request out into two... but would it be possible for fat and protein entries to be captured as a single entry with text to show amount over defined duration?

Thanks!

**Comment by Jon-b-m** [comment]
Agreed. Carbs shouldn't be saved before confirming. The Fat and Protein carb equivalents are all saved with the same ID, so you only need to tap delete on one of those entries.

**Comment by NightScoutForSeb** [comment]
I just found this same issue when testing the new bolus calculator in dev. I pretended that I wanted to change the carb count via the Apple Watch carb entry screen, and couldn't. I had to go into the phone app and delete the entry, then re-add what I wanted to.  I was also not able to edit the time for the existing carb entries in the phone app history, like I am able to in Loop - it also happens that my child ends up eating less than anticipated, and I cannot edit the entry in iAPS. If he eats more, I can always have him add another carb entry. Eating less is a common scenario for parents with smaller Type 1 children in the Loop community. It is really nice to be able to 'fix' the carb entry historically, to ensure the algorithm isn't too aggressive, causing lows later on.
